core/src/main/java/com/opensymphony/xwork2/config/providers/XmlConfigurationProvider.java

```diff
@@ -439,28 +445,16 @@ protected void addAction(Element actionElement, PackageConfig.Builder packageCon
         String name = actionElement.getAttribute("name");
         String className = actionElement.getAttribute("class");
         //methodName should be null if it's not set
-        String methodName = StringUtils.trimToNull(actionElement.getAttribute("method"));
+        String methodName = trimToNull(actionElement.getAttribute("method"));
         Location location = DomHelper.getLocationObject(actionElement);
 
         if (location == null) {
             LOG.warn("Location null for {}", className);
         }
 
-        // if there isn't a class name specified for an <action/> then try to
-        // use the default-class-ref from the <package/>
-        if (StringUtils.isEmpty(className)) {
-            // if there is a package default-class-ref use that, otherwise use action support
-           /* if (StringUtils.isNotEmpty(packageContext.getDefaultClassRef())) {
-                className = packageContext.getDefaultClassRef();
-            } else {
-                className = ActionSupport.class.getName();
-            }*/
-
-        } else {
-            if (!verifyAction(className, name, location)) {
-                LOG.error("Unable to verify action [{}] with class [{}], from [{}]", name, className, location);
-                return;
-            }
+        if (!className.isEmpty() && !verifyAction(className, name, location)) {
```

Comment:
Maybe we can remove the unused param `name` from verifyAction L:490

Sonar says: Parameter 'name' is never used

Reply:
There's multiple methods in this class with unused params - I've kept them to maintain compatibility with potential subclasses. Perhaps something to revisit when releasing the next major version.

Reply:
@sepe81 I'm right now creating a task to mark a given code `@Deprecated` and then another task (targeting major/minor release) to remove the code. I found such approach more useful and more informative for the users :)